File: Data Structures & Algorithms/majority-element/submission-0.go
func majorityElement(nums []int) int {
    var freq int = 0
    var ans int 
    for _ , n := range nums {
        if freq == 0 {
            ans = n 
        }
        if n == ans {
            freq += 1
        }else {
            freq -= 1
        }
    }
    return ans
}

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.cpp
class Solution {
public:
    vector<int> productExceptSelf(vector<int>& nums) {
        int n = nums.size() ;
        vector<int> p(n+1,1), s(n+1,1);
        for(auto i = 0 ; i < n; ++i){
            p[i+1] = p[i] * nums[i];
        }
        for(auto i = n - 1; i >= 0; --i){
            s[i] = s[i+1] * nums[i];
        }
        vector<int>ans (n);
        for(auto i = 0 ; i < n ; ++i){
            ans[i] = p[i] * s[i+1];
        }
        return ans;
    }
};
